app/services/conversion/TransitAccompanyingDocumentConversionService.scala

```diff
@@ -114,4 +119,80 @@ class TransitAccompanyingDocumentConversionService @Inject() (referenceData: Ref
       result => result
     )
   }
+
+  def fromP5ToViewModel(
+    ie029: IE029,
+    ie015: IE015
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[ValidationResult[TransitAccompanyingDocumentP5TransitionPDF]] = {
+
+    val countriesFuture: Future[ValidationResult[Seq[Country]]] = referenceDataP5.getList[Seq[Country]]("CountryCodesForAddress")
+    val additionalInfoFuture: Future[ValidationResult[Seq[AdditionalInformation]]] =
+      referenceDataP5.getList[Seq[AdditionalInformation]]("AdditionalInformation")
+    val kindsOfPackageFuture: Future[ValidationResult[Seq[KindOfPackage]]] = referenceDataP5.getList[Seq[KindOfPackage]]("KindOfPackages")
+    val previousDocumentTypesFuture: Future[ValidationResult[Seq[PreviousDocumentTypes]]] =
+      referenceDataP5.getList[Seq[PreviousDocumentTypes]]("PreviousDocumentType")
+    val supportingDocumentTypesFuture: Future[ValidationResult[Seq[SupportingDocumentTypes]]] =
+      referenceDataP5.getList[Seq[SupportingDocumentTypes]]("SupportingDocumentType")
+    val transportDocumentTypesFuture: Future[ValidationResult[Seq[TransportDocumentTypes]]] =
+      referenceDataP5.getList[Seq[TransportDocumentTypes]]("TransportDocumentType")
+    val circumstanceIndicatorsFuture: Future[ValidationResult[Seq[CircumstanceIndicator]]] =
+      referenceDataP5.getList[Seq[CircumstanceIndicator]]("SpecificCircumstanceIndicatorCode")
+    val customsOfficeFuture: Future[ValidationResult[Seq[CustomsOffice]]] = referenceDataP5.getList[Seq[CustomsOffice]]("customsOffices")
+//    val controlResultFuture: Future[ValidationResult[Seq[ControlResult]]] =
+//      referenceDataP5.getList[Seq[ControlResult]]("controlResult")
+
+    for {
+      countryCodesForAddress  <- countriesFuture
+      additionalInfo          <- additionalInfoFuture
+      kindsOfPackage          <- kindsOfPackageFuture
+      previousDocumentTypes   <- previousDocumentTypesFuture
+      supportingDocumentTypes <- supportingDocumentTypesFuture
+      transportDocumentTypes  <- transportDocumentTypesFuture
+      circumstanceIndicators  <- circumstanceIndicatorsFuture
+      customsOffice           <- customsOfficeFuture
+//      controlResult <- controlResultFuture
+    } yield (
```

Comment:
Just a similar comment on the yield, can we move the logic up into the for comp and maybe split into a separate function if it looks to big